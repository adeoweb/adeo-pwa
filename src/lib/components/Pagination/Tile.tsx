import React, { useCallback, useMemo, FunctionComponent } from 'react';

interface ITileProps {
    isActive: boolean;
    number: number;
    onClick: (page: number) => void;
    getPageUrl: (page: number) => string;
}

const Tile: FunctionComponent<ITileProps> = props => {
    const { isActive, number, onClick, getPageUrl } = props;

    const handleClick = useCallback(
        event => {
            event.preventDefault();
            onClick(number);
        },
        [onClick, number]
    );

    const pageUrl = useMemo(() => getPageUrl(number), [getPageUrl, number]);

    return (
        <li className={`page-item ${isActive ? 'active' : ''}`}>
            <a className="page-link" href={pageUrl} onClick={handleClick}>
                {number}
                {isActive && <span className="sr-only">(current)</span>}
            </a>
        </li>
    );
};

export default Tile;
