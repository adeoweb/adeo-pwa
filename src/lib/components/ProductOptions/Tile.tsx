import React, { FunctionComponent } from 'react';

import { useTile } from '@magento/peregrine/lib/talons/ProductOptions/useTile';

import { TConfigurableProductOptionsValues } from 'src/lib/types/graphql/Product';

type TTileProps = {
    isSelected: boolean;
    item: TConfigurableProductOptionsValues;
    onClick: (selection: number) => void;
};

const Tile: FunctionComponent<TTileProps> = ({
    isSelected,
    item: { label, value_index },
    onClick
}) => {
    const talonProps = useTile({
        onClick,
        value_index
    });

    const { handleClick } = talonProps;

    return (
        <li className={isSelected ? 'active' : ''}>
            <button onClick={handleClick}>{label}</button>
        </li>
    );
};

export default Tile;
