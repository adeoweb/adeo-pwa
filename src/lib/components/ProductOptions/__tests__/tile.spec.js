import React from 'react';
import testRenderer from 'react-test-renderer';

import Tile from '../Tile';

const defaultProps = {
    item: {
        label: 'label',
        value_index: 0
    }
};

test('renders a Tile correctly', () => {
    const component = testRenderer.create(<Tile {...defaultProps} />);

    expect(component.root.findByType('li').props.className).not.toContain(
        'active'
    );

    expect(component).toMatchSnapshot();
});

test('appends "active" to className if isSelected is true', () => {
    const props = {
        ...defaultProps,
        isSelected: true
    };

    const component = testRenderer.create(<Tile {...props} />);

    expect(component.root.findByType('li').props.className).toContain('active');

    expect(component).toMatchSnapshot();
});
