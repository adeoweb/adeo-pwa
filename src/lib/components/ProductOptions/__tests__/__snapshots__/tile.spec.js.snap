// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appends "active" to className if isSelected is true 1`] = `
<li
  className="active"
>
  <button
    onClick={[Function]}
  >
    label
  </button>
</li>
`;

exports[`renders a Tile correctly 1`] = `
<li
  className=""
>
  <button
    onClick={[Function]}
  >
    label
  </button>
</li>
`;
